libraries/AP_FETtecOneWire/AP_FETtecOneWire.cpp

```diff
@@ -175,48 +175,49 @@ void AP_FETtecOneWire::update()
         if (c == nullptr) {
             continue;
         }
-        motorpwm[i] = c->get_output_pwm();
+        _motorpwm[i] = c->get_output_pwm();
     }
 
     uint16_t requestedTelemetry[MOTOR_COUNT_MAX] = {0};
-    TelemetryAvailable = ESCsSetValues(motorpwm, requestedTelemetry, MOTOR_COUNT_MAX, TLM_request_type);
-    if (TelemetryAvailable != -1) {
+    _telem_avail = ESCsSetValues(_motorpwm, requestedTelemetry, MOTOR_COUNT_MAX, _telem_req_type);
+    const uint32_t now = AP_HAL::millis();
+    if (_telem_avail != -1) {
         for (uint8_t i = 0; i < MOTOR_COUNT_MAX; i++) {
-            completeTelemetry[i][TelemetryAvailable] = requestedTelemetry[i];
-            completeTelemetry[i][5]++;
+            _telemetry[i][_telem_avail] = requestedTelemetry[i];
+            _telemetry[i][5]++;
+            _telemetry[i][6] = now;
         }
     }

```

Comment:
You need to call update_rpm() somewhere about here

Reply:
Thanks!!!

Reply:
done, thanks for the review!